File: Detect_Capital.js
/**
 * @param {string} word
 * @return {boolean}
 */
function detectCapitalUse(word) {
  let counterUppercaseLetters = 0;
  
  for (let i = 0; i < word.length; i++) {
    if (word[i] === word[i].toUpperCase()) {
      counterUppercaseLetters++;
    }
  }
  
  return counterUppercaseLetters === 0 ||
         counterUppercaseLetters === word.length ||
         (counterUppercaseLetters === 1 && word[0] === word[0].toUpperCase());
}

File: Detect_Capital.java
class Solution {
public static boolean detectCapitalUse(String word) {
    int counterUppercaseLetters = 0;

    for (int i = 0; i < word.length(); i++) {
        if (word.charAt(i) == Character.toUpperCase(word.charAt(i))) {
            counterUppercaseLetters++;
        }
    }

    return counterUppercaseLetters == 0 ||
            counterUppercaseLetters == word.length() ||
            (counterUppercaseLetters == 1 && word.charAt(0) == Character.toUpperCase(word.charAt(0)));
}
}